core/src/main/scala/kafka/utils/ShutdownableThread.scala

```diff
@@ -63,6 +65,8 @@ abstract class ShutdownableThread(val name: String, val isInterruptible: Boolean
         doWork()
       }
     } catch{
+      case fatalExitException: FatalExitException =>
```

Comment:
FatalExitException will never be catched here because processFetchRequest catches all exceptions.
Also addPartition should be checked.